test/test_piece_picker.cpp

```diff
@@ -753,6 +753,19 @@ TORRENT_TEST(sequential_download)
 			, i % blocks_per_piece));
 }
 
+TORRENT_TEST(sequential_range_download)
+{
+	// test sequential range download
+	auto p = setup_picker("7654321", "       ", "", "");
+	p->set_sequential_range(piece_index_t(1), piece_index_t(5));
+	auto picked = pick_pieces(p, "*******", 7 * blocks_per_piece, 0, nullptr
+		, piece_picker::sequential, empty_vector);
+	TEST_CHECK(int(picked.size()) == 5 * blocks_per_piece);
+	for (int i = 0; i < int(picked.size()); ++i)
+		TEST_CHECK(picked[std::size_t(i)] == piece_block(piece_index_t((i + 4) / blocks_per_piece)
```

Comment:
this "4" is `blocks_per_piece`, right? I think it would be much clearer what's going on if you used the named constant instead.

Reply:
This 4 actually represents the second piece in the piece map (at index 1) - not blocks per piece.

Reply:
so maybe even better if it would be `1 * blocks_per_piece`

Reply:
Ok yes.